tests/test_examples.py

```diff
@@ -28,54 +28,53 @@ def _fake_data():
 
-        # Should be able to call get_data without being asked to authenticate in tests
-        X, y = dataset.get_data()
+    # Should be able to call get_data without being asked to authenticate in tests
+    X, y = dataset.get_data()
 
-    @mock.patch.object(TimeSeriesDataset, "get_data", return_value=_fake_data())
-    def test_notebooks(self, _mocked_method):
-        """
-        Ensures all notebooks will run without error
-        """
-        repo_root = os.path.join(os.path.dirname(__file__), "..")
-        notebooks = glob.glob(os.path.join(repo_root, "examples", "*.ipynb"))
 
-        logger.info(f"Found {len(notebooks)} notebooks to test.")
+@mock.patch("gordo_components.builder.azure_utils.MlflowClient", autospec=True)
+@mock.patch.object(TimeSeriesDataset, "get_data", return_value=_fake_data())
+def test_notebooks(MockDataset, MockClient):
+    """
+    Ensures all notebooks will run without error
+    """
+    repo_root = os.path.join(os.path.dirname(__file__), "..")
+    notebooks = glob.glob(os.path.join(repo_root, "examples", "*.ipynb"))
 
-        for notebook in notebooks:
+    logger.info(f"Found {len(notebooks)} notebooks to test.")
 
-            logger.info(f"Testing notebook: {os.path.basename(notebook)}")
+    for notebook in notebooks:
 
-            with open(notebook) as f:
-                nb = nbformat.read(f, as_version=4)
-                exporter = PythonExporter()
-                source, _meta = exporter.from_notebook_node(nb)
+        logger.info(f"Testing notebook: {os.path.basename(notebook)}")
 
-                with tempfile.TemporaryDirectory() as tmpdir:
-                    with open(os.path.join(tmpdir, "tmpmodule.py"), "w") as f:
-                        f.writelines(source)
-                    with open(os.path.join(tmpdir, "__init__.py"), "w") as f:
-                        f.write("from .tmpmodule import *")
+        with open(notebook) as f:
+            nb = nbformat.read(f, as_version=4)
+            exporter = PythonExporter()
+            source, _meta = exporter.from_notebook_node(nb)
 
-                    # Import this module to 'run' the code
-                    module_dir = os.path.join(tmpdir, "..")
-                    sys.path.insert(0, module_dir)
+            with tempfile.TemporaryDirectory() as tmpdir:
+                with open(os.path.join(tmpdir, "tmpmodule.py"), "w") as f:
+                    f.writelines(source)
+                with open(os.path.join(tmpdir, "__init__.py"), "w") as f:
+                    f.write("from .tmpmodule import *")
 
-                    importlib.import_module(os.path.basename(tmpdir), ".")
+                # Import this module to 'run' the code
+                module_dir = os.path.join(tmpdir, "..")
+                sys.path.insert(0, module_dir)
 
-                    sys.path.remove(module_dir)
+                importlib.import_module(os.path.basename(tmpdir), ".")
+
+                sys.path.remove(module_dir)
```

Comment:
Again, I think the PR is enough as it is, migrating tests to pytest is probably better left for a different PR.

Reply:
I'll pull this out to another PR.